modules/semanticsearch/src/main/kotlin/com/theagilemonkeys/ellmental/semanticsearch/SemanticSearch.kt

```diff
@@ -35,18 +36,19 @@ class SemanticSearch {
      * a ranked list of results.
      *
      * @param text The text to be used as reference for semantic search.
+     * @param itemsLimit The maximum number of results to return.
      * @return A list of semantically similar texts ranked by semantic distance (the closest first).
      */
-    suspend fun search(text: String): SearchOutput {
+    suspend fun search(text: String, itemsLimit: Int): SearchOutput {
```

Comment:
```suggestion
    suspend fun search(text: String, itemsLimit?: Int): SearchOutput {
```

Can we make the itemsLimit optional? Perhaps also adding a default of 5 or 10